src/components/KModal/KModal.vue

```diff
@@ -141,6 +274,54 @@ export default defineComponent({
       emit('proceed')
     }
 
+    // only bother with this if we have a src for the header image and a dismiss button with 'auto' for the dismissButtonShade
+    const shouldComputeImageDarkness = computed(() => {
+      return !!(props.headerImageSrc && props.enableDismiss && props.dismissButtonShade === 'auto')
+    })
+    const computedImageDarkness = ref('none')
+    const getComputedImageDarkness = () => {
+      // don't bother unless the dialog is visible, the canvas has been initialized, and
+      // we should
+      if (props.isVisible && canvas.value && shouldComputeImageDarkness.value) {
+        const ctx = canvas.value
+        const img = document.createElement('img')
+        img.src = props.headerImageSrc
+
+        img.style.display = 'none'
+        document.body.appendChild(img)
+
+        img.onload = function() {
+          imageStats.value.height = img.naturalHeight
+          imageStats.value.width = img.naturalWidth
+          const width = imageStats.value.width
+          const height = imageStats.value.height
+          ctx.drawImage(img, 0, 0)
+          const imageData = ctx.getImageData(0, 0, width, height)
+
+          const data = imageData.data
+          let r, g, b, avg
+          let colorSum = 0
+
+          for (let x = 0, len = data.length; x < len; x += 4) {
+            r = data[x]
+            g = data[x + 1]
+            b = data[x + 2]
+
+            avg = Math.floor((r + g + b) / 3)
+            colorSum += avg
+          }
+
+          const brightness = Math.floor(colorSum / (width * height))
+
+          if (brightness < 127.5) {
+            computedImageDarkness.value = 'dark'
+          } else {
+            computedImageDarkness.value = 'light'
+          }
+        }
+      }
+    }
```

Comment:
This seems expensive just to calculate the close button color for a component. If we were utilizing this for something more "important" than a close button, I could see the value; however, I kind of want to recommend setting it by default to dark, and then a prop can be used to change it to light if needed.

Nice job, but I just think it's too "smart" for a close button.

Reply:
Removed